packages/playground/remote/src/lib/boot-playground-remote.ts

```diff
@@ -232,6 +239,26 @@ export async function bootPlaygroundRemote() {
 		throw e;
 	}
 
+	/**
+	 * If the browser is online we download the WordPress assets asynchronously to speed up the boot process.
+	 * Missing assets will be fetched on demand from the Playground server until they are downloaded.
+	 *
+	 * If the browser is offline, we backfill WordPress assets synchronously
+	 * from cache to ensure Playground is fully functional before boot finishes.
+	 */
+	if (window.navigator.onLine) {
+		wpFrame.addEventListener('load', () => {
+			webApi.backfillStaticFilesRemovedFromMinifiedBuild();
+		});
+	} else {
+		webApi.setProgress({
```

Comment:
```suggestion
		// Note this will run even if the static files are already in place, e.g. when running
		// a non-minified build or an offline site. It doesn't seem like a big problem worth introducing
		// a new API method like `webApi.needsBackfillingStaticFilesRemovedFromMinifiedBuild().
		webApi.setProgress({
```